daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/DaffodilParseOutputStreamContentHandler.scala

```diff
@@ -116,9 +256,19 @@ class DaffodilParseOutputStreamContentHandler(out: OutputStream, pretty: Boolean
       }
     }
     writer.write("</")
-    writer.write(qName)
+    outputTagName(uri, localName, qName)
     writer.write(">")
     outputNewlineStack.pop()
+
+    // throw out current prefix mapping context as we're done with it
+    if (!activePrefixMappingContextStack.isEmpty) activePrefixMappingContextStack.pop
```

Comment:
Shouldn't this condition always be true? I think we always push onto this stack when startElement is called, and so there should always be an element to pop when endELement is called. I guess the only time that wouldn't be true is if the XMLReader messed up and called endElement without calling a matching startElement but that should be an error. Perhaps this condition that the stack isn't empty should just be changed to an Assert.invariant?